editor/src/dispatcher.rs

```diff
@@ -104,9 +117,9 @@ impl Dispatcher {
 					self.buffered_queue = Some(std::mem::take(&mut self.message_queues));
 				}
 				Message::EndBuffer(render_metadata) => {
-					// The buffered vec is added before the metadata messages, because the end of the vec is processed first
+					// Assign the message queue to the currently buffered queue
 					if let Some(buffered_queue) = self.buffered_queue.take() {
-						self.message_queues.extend(buffered_queue);
+						self.message_queues = buffered_queue;
```

Comment:
Is there a reason why the buffered queue replaces the current message queues? I think it could be possible for there to still be messages in self.message_queues, which would be lost here.

Reply:
The current message queues will always be empty (except for the end buffer event), since they are cleared by the start buffer event and all other messages are added directly to the buffer.